webapp/src/Entity/ContestProblem.php

```diff
@@ -193,17 +194,27 @@ public function getApiColor(): ?string
         return Utils::convertToColor($this->getColor());
     }
 
-    public function setLazyEvalResults(?int $lazyEvalResults): ContestProblem
+    public function setLazyEvalResults(int $lazyEvalResults): ContestProblem
     {
-        $this->lazyEvalResults = $lazyEvalResults === 0 ? null : $lazyEvalResults;
+        $this->lazyEvalResults = $lazyEvalResults;
         return $this;
     }
 
-    public function getLazyEvalResults(): ?int
+    public function getLazyEvalResults(): int
     {
         return $this->lazyEvalResults;
     }
 
+    public function determineOnDemand(int $config_lazy_eval_results) {
+        if ($this->lazyEvalResults === DJS::EVAL_DEMAND) {
+            return true;
+        }
+        if ($this->lazyEvalResults === DJS::EVAL_DEFAULT && $config_lazy_eval_results) {
```

Comment:
Shouldn't this be
```suggestion
        if ($this->lazyEvalResults === DJS::EVAL_DEFAULT && $config_lazy_eval_results === DJS:: EVAL_DEMAND) {
```
?

Reply:
Yes